src/IotTelemetrySimulator/EventHubSender.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Microsoft.Azure.EventHubs;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+
+namespace IotTelemetrySimulator
+{
+    public class EventHubSender : ISender
+    {
+        const int WaitTimeOnIotHubError = 5_000;
+
+        private string deviceId;
+        private RunnerConfiguration config;
+        private EventHubClient eventHubClient;
+        private Dictionary<string, object> variableValues;
+
+        public EventHubSender(string deviceId, RunnerConfiguration config)
+        {
+            this.deviceId = deviceId;
+            this.config = config;
+        }
+
+        public async Task OpenAsync()
+        {
+            eventHubClient = EventHubClient.CreateFromConnectionString(config.EventHubConnectionString);
+        }
+
+        public async Task SendMessageAsync(RunnerStats stats, CancellationToken cancellationToken)
+        {
+            const int MaxAttempts = 3;
+            var msg = CreateMessage();
+
+            for (var attempt = 1; attempt <= MaxAttempts; ++attempt)
+            {
+                try
+                {
+                    await eventHubClient.SendAsync(msg);
+                    stats.IncrementMessageSent();
```

Comment:
We are repeating the `stats.IncrementMessageSent()` and `stats.IncrementSendTelemetryErrors()` in both senders. Wonder if we can share this code in a common class (i.e. composition pattern)

Reply:
I added an abstract base class SenderBase to reduce code duplication